File: src/main/java/com/murex/tbw/domain/Novel.java
package com.murex.tbw.domain;

public class Novel implements Book {

    private final String name;
    private final double price;
    private final Author author;

    public Novel(String name, double price, Author author) {
        this.name = name;
        this.price = price;
        this.author = author;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public double getPrice() {
        return price;
    }

    @Override
    public Author getAuthor() {
        return author;
    }


}

File: src/main/cpp/lib/Novel.cpp
/*******************************************************************************
 *
 * Copyright (c) {2003-2019} Murex S.A.S. and its affiliates.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the MIT License
 * which accompanies this distribution, and is available at
 * https://opensource.org/licenses/MIT
 *
 *******************************************************************************/

#include "domain/book/Novel.h"

domain::book::Novel::Novel(const std::string& name, double price, const Author& author, country::Language language,
                           const std::vector<Genre>& genres): name_(name)
                                                              , price_(price)
                                                              , author_(author)
                                                              , language_(language)
                                                              , genres_(genres)
{
}
